src/allmydata/testing/__init__.py

```diff
@@ -0,0 +1,18 @@
+import os
+from typing import Optional
+
+
+def foolscap_only_for_integration_testing() -> Optional[bool]:
+    """
+    Return whether HTTP storage protocol has been disabled / Foolscap
+    forced, for purposes of integration testing.
+
+    This is determined by the __TAHOE_INTEGRATION_FORCE_FOOLSCAP environment
+    variable, which can be 1, 0, or not set, corresponding to results of
+    ``True``, ``False`` and ``None`` (i.e. default).
```

Comment:
Does it really need to be a tri-state?  Isn't the CI configuration only exercising the True and False cases (neither of which is the case you'd want to use in production)?

Reply:
There's the "I'm running tests locally" case. And there is the issue that, since it's not set in production, one ought to run tests without it set too? Which this branch doesn't. I'll think about it...